_studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp

```diff
@@ -0,0 +1,1196 @@
+// Copyright (c) 2017-2020 Intel Corporation
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#include "mfx_common.h"
+
+#if defined(MFX_ENABLE_AV1_VIDEO_DECODE)
+
+#include "mfx_session.h"
+#include "mfx_av1_dec_decode.h"
+
+#include "mfx_task.h"
+
+#include "mfx_common_int.h"
+#include "mfx_common_decode_int.h"
+#include "mfx_vpx_dec_common.h"
+
+#include "mfx_umc_alloc_wrapper.h"
+
+#include "umc_av1_dec_defs.h"
+#include "umc_av1_frame.h"
+#include "umc_av1_utils.h"
+
+#include "libmfx_core_hw.h"
+
+#include "umc_av1_decoder_va.h"
+
+#include <algorithm>
+
+#include "vm_sys_info.h"
+
+#if defined(MFX_VA)
+#include "umc_h265_va_supplier.h"
+#if defined(MFX_ENABLE_CPLIB) || !defined(MFX_PROTECTED_FEATURE_DISABLE)
+#include "umc_va_linux_protected.h"
+#endif
+#endif
+
+inline
+mfxU32 CalculateAsyncDepth(eMFXPlatform platform, mfxVideoParam *par)
+{
+    mfxU32 asyncDepth = par ? par->AsyncDepth : 0;
+    if (!asyncDepth)
+    {
+        asyncDepth = (platform == MFX_PLATFORM_SOFTWARE) ? vm_sys_info_get_cpu_num() : MFX_AUTO_ASYNC_DEPTH_VALUE;
+    }
+
+    return asyncDepth;
+}
+
+namespace MFX_VPX_Utility
+{
+    inline
+    mfxU16 MatchProfile(mfxU32 fourcc)
```

Comment:
nameless argument is enough to mute warning^
```
mfxU16 MatchProfile(mfxU32)
```

Reply:
sure.